### DISCUSSION TITLE
FirebaseError: projectId must be a string in FirebaseApp.options

### DISCUSSION THREAD

**Discussion by avg-developer** [vercel/vercel#4657] [Help] [answered]
I just made a fresh deploy of my React firebase app i get this after vercel deploy is done
Locally is working fine, i've all setted. 

This is how my firebase config file looks like.

`import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/database';
import 'firebase/firestore';
import 'firebase/storage';

import { createStore, combineReducers } from 'redux';
import { firebaseReducer } from 'react-redux-firebase';
import { createFirestoreInstance, firestoreReducer } from 'redux-firestore'; // <- needed if using firestore

import config from './fbConfig';

// react-redux-firebase config
const rrfConfig = {
  userProfile: 'users', // where profiles are stored in database
  presence: 'presence', // where list of online users is stored in database
  sessions: 'sessions', // where list of user sessions is stored in database (presence must be enabled)
  useFirestoreForProfile: true // Firestore for Profile instead of Realtime DB
};

// Initialize firebase instance
firebase.initializeApp(config);
firebase.firestore().enablePersistence();

// Initialize other services on firebase instance
firebase.firestore(); // <- needed if using firestore

// Add firebase to reducers
const rootReducer = combineReducers({
  firebase: firebaseReducer,
  firestore: firestoreReducer // <- needed if using firestore
});

// Create store with reducers and initial state
const initialState = {};
export const store = createStore(rootReducer, initialState);

export const rrfProps = {
  firebase,
  config: rrfConfig,
  dispatch: store.dispatch,
  createFirestoreInstance
};
`

**Comment by markozxuu** [comment]
Hi @avg-developer 👋  Could you give us more information. a client-side error would help much more

**Reply by valenciaHQ** [reply]
Solved using env variables at Vercel Platform. Thanks!

**Comment by valenciaHQ** [comment, marked as answer]
This should be closed. Solved using env variables at Vercel Platform. Thanks!
